src/storage-types/src/stats.rs

```diff
@@ -92,15 +92,15 @@ impl RelationPartStats<'_> {
         result.may_contain(Datum::True) || result.may_fail()
     }
 
-    fn json_spec<'a>(len: usize, stats: &'a JsonStats, arena: &'a RowArena) -> ResultSpec<'a> {
+    fn json_spec<'a>(len: usize, stats: JsonStats, arena: &'a RowArena) -> ResultSpec<'a> {
```

Comment:
The cost of this change is some extra arena allocations farther down, but I don't immediately see the benefit -- could you talk me through it?

Reply:
Cool - if we're expecting this to change I won't worry too much about it! A few allocations for stats is not the end of the world.